funsor/gaussian.py

```diff
@@ -254,101 +251,27 @@ def eager_reduce(self, op, reduced_vars):
         return None  # defer to default implementation
 
 
-@eager.register(Binary, Op, Gaussian, Number)
```

Comment:
I've moved `log_density` tracking from `Gaussian` to `Joint`, which simplifies the code here.

Reply:
This looks great now!